dsc/docs-conceptual/dsc-2.0/concepts/class-based-resources.md

```diff
@@ -680,22 +749,75 @@ an invalid value and how the value was invalid.
 
 If a property of the class-based DSC Resource has a list of valid values it accepts, define it as an
-[enum property][20] instead of using the **ValidateSet** attribute.
+[enum property][24] instead of using the **ValidateSet** attribute.
 
 This gives users a better error message and makes maintenance easier if you use those values
 anywhere else in your module. Instead of having to update every cmdlet or check for that property,
 you can update the enum definition.
 
+Remember that enums are value type properties. They default to `0` when an instance of the
+class-based DSC Resource is created. If you set an explicit default value for an enum property, that
+enum can't be unmanaged. If a user doesn't specify the property, the DSC Resource will behave the
+same as if the user had explicitly specified the default value.
+
+To support unmanaged enum properties, make sure not to define a label for the `0` value. By default,
+if you don't specify the integer value for any labels when using the `enum` statement, the first
+declared label has a value of `0`. Instead, define the first label with a value of `1` (or any
+higher value).
+
+For example:
+
+```powershell
+enum OptionalSetting {
+    FirstOption = 1
+    SecondOption
+    ThirdOption
+}
+```
+
+Then, in your **Test** and **Set** methods, you can ignore the enum property if the value is `0`.
+In **Test**, if the enum property is `0`, ignore the current state of the DSC Resource on the
+system. Don't report the DSC Resource as out of desired state if the current state is a valid value.
+In **Set**, make sure your logic for modifying system state ignores the enum property if it's `0`.
+
+> [!NOTE]
+> This method for implementing enum properties that can be unmanaged only works when the underlying
+> value of the enum labels doesn't matter or when `0` isn't a valid underlying value.
+
+### Explicitly define default values for value type properties
+
+When authoring a class-based DSC Resource, ensure that your properties specify correct default
+values for every value type property. You'll need to ensure your default values stay in sync with
+those of the component your DSC Resource manages.
+
+Document the behavior of these properties for your users. Make sure that your documentation explains
+that these properties will be set to their default value if the user doesn't specify them when using
+the DSC Resource.
+
+### Ensure the DSC Resource respects unmanaged reference type properties
+
+Because reference type properties initialize to `$null`, unlike value type properties, a DSC
+Resource can distinguish between whether a reference type property was specified or not. Make sure
+that your DSC Resources ignore unmanaged reference type properties in the **Test** and **Set**
+methods.
+
+In **Test**, if the reference type property is `$null`, ignore the current state of the DSC Resource
+on the system. Don't report the DSC Resource as out of desired state if the current state isn't
+`$null`.
```

Comment:
The double negatives here are confusing. Did you mean to say: "if the current state IS $null"?

Reply:
No. Normally, if the current state and desired state don't match, you would return `$false` and consider that the system is out of the desired state.

If a reference value property is `$null`, the user did not explicitly choose to manage it. In that case, treat the property as being in the desired state regardless of the value for that property in the current state. In this case, a non-null value in the current state doesn't indicate a Test method failure because the user did not specify how to manage the property.